dref/models.py

```diff
@@ -0,0 +1,568 @@
+        null=True, blank=True
+    )
+    originator_email = models.CharField(
+        verbose_name=_('originator email'), max_length=255,
+        null=True, blank=True
+    )
+    originator_title = models.CharField(
+        verbose_name=_('originator title'), max_length=255,
+        null=True, blank=True
+    )
+    originator_phone_number = models.CharField(
+        verbose_name=_('originator phone number'), max_length=100,
+        null=True, blank=True
+    )
+    human_resource = models.TextField(
+        blank=True, null=True,
+        verbose_name=_('human resource'),
+        help_text=_('how many volunteers and staff involved in the response?')
+    )
+    surge_personnel_deployed = models.TextField(
+        blank=True, null=True,
+        verbose_name=_('surge personnel deployed'),
+        help_text=_('Will a Surge personnel be deployed?')
+    )
+    logistic_capacity_of_ns = models.TextField(
+        blank=True, null=True,
+        verbose_name=_('logistic capacity of ns'),
+        help_text=_('what is the logistics capacity of the National Society?')
+    )
+    safety_concerns = models.TextField(
+        blank=True, null=True,
+        verbose_name=_('safety concerns'),
+        help_text=_('Are there any safety/security concerns which may impact the implementation of this operation?')
+    )
+    pmer = models.TextField(
+        blank=True, null=True,
+        verbose_name=_('pmer'),
+        help_text=_('Does the NS have PMER capacity?')
+    )
+    communication = models.TextField(
+        blank=True, null=True,
+        verbose_name=_('organization'),
+        help_text=_('Does the NS have Communications capacity?')
+    )
+    event_map = models.ForeignKey(
+        'DrefFile', on_delete=models.SET_NULL,
+        null=True, blank=True,
+        verbose_name=_('event map'),
+        related_name='event_map_dref'
+    )
+    images = models.ManyToManyField(
+        'DrefFile', blank=True, null=True,
+        verbose_name=_('images'),
+        related_name='image_dref'
+    )
+    budget_file = models.ForeignKey(
+        'DrefFile', on_delete=models.SET_NULL,
+        null=True, blank=True,
+        verbose_name=_('budget file'),
+        related_name='budget_file_dref'
+    )
+    budget_file_preview = models.FileField(
+        verbose_name=_('budget file preview'),
+        null=True, blank=True,
+        upload_to='dref/images/'
+    )
+    cover_image = models.ForeignKey(
+        'DrefFile', on_delete=models.SET_NULL,
+        null=True, blank=True,
+        verbose_name=_('cover image'),
+        related_name='cover_image_dref'
+    )
+
+    class Meta:
+        verbose_name = _('dref')
+        verbose_name_plural = _('drefs')
+
+    def save(self, *args, **kwargs):
+        # TODO: Need to remove this if not required later
+        if self.date_of_approval:
+            self.status = Dref.Status.COMPLETED
+        elif not self.date_of_approval:
+            self.status = Dref.Status.IN_PROGESS
```

Comment:
Is this still required?

Also we can use single conditional here.
```suggestion
        self.status = Dref.Status.COMPLETED if self.date_of_approval else Dref.Status.IN_PROGESS
```